### PR TITLE
Angular: Make angular work with styles.bundleName

### PR DESCRIPTION
**Pull request by ThibaudAV**
Issue: https://github.com/storybookjs/storybook/issues/17711

## What I did

Make angular work with styles.bundleName inside angular.json

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

Tested with https://github.com/SunilManthenaG01/storybook-inline-scss-angular- from issue

- [ ] Test with angular 13 ?

### CODE REVIEW

File: app/angular/src/server/angular-cli-webpack-12.2.x.js [hunk 1]
```diff
@@ -46,12 +46,9 @@ exports.getWebpackConfig = async (baseConfig, { builderOptions, builderContext }
 
   /**
    * Merge baseConfig Webpack with angular-cli Webpack
+   * Keep all entry from angular cli except main. Add storybook entry to man key
```
Comment (kroeder) [inline app/angular/src/server/angular-cli-webpack-12.2.x.js:49]: > Add storybook entry to >>>man<<< key

typo?

File: app/angular/src/server/angular-cli-webpack-12.2.x.js [hunk 2]
```diff
@@ -46,12 +46,9 @@ exports.getWebpackConfig = async (baseConfig, { builderOptions, builderContext }
...
    */
-  const entry = [
-    ...baseConfig.entry,
-    ...(cliConfig.entry.styles ?? []),
-    ...(cliConfig.entry.polyfills ?? []),
-  ];
+  const entry = { ...cliConfig.entry, main: baseConfig.entry };
```
Comment (kroeder) [inline app/angular/src/server/angular-cli-webpack-12.2.x.js:51]: Is it correct, that both are called `entry`?

```
{
  ...cliConfig.entry,
  main: baseConfig.entry
}
```

The object looks like this then, right?

```
{
  styles: [...],
  polyfilles: [...],
  main: [...],
}
```

File: app/angular/src/server/angular-cli-webpack-13.x.x.js [hunk 3]
```diff
@@ -51,12 +51,9 @@ exports.getWebpackConfig = async (baseConfig, { builderOptions, builderContext }
 
   /**
    * Merge baseConfig Webpack with angular-cli Webpack
+   * Keep all entry from angular cli except main. Add storybook entry to man key
```
Comment (kroeder) [inline app/angular/src/server/angular-cli-webpack-13.x.x.js:54]: > Add storybook entry to >>>man<<< key

typo?

**Review by kroeder — APPROVED** [review]
LGTM, just left questions

File: app/angular/src/server/angular-cli-webpack-12.2.x.js (hunk 2 above)
Comment (ThibaudAV) [reply app/angular/src/server/angular-cli-webpack-12.2.x.js:51]: Yes. And allow angular to add more entry arrays

**Comment by shilman** [comment]
@kroeder @ThibaudAV looks like build is failing. can you please take a look?

https://app.circleci.com/pipelines/github/storybookjs/storybook/24995/workflows/398e6c07-e5de-4bb1-b0bd-c5a0acd12733/jobs/373938

**Comment by ndelangen** [comment]
Hey @kroeder is this something we'll be able to merge soon?

**Pull request closed without merging**